### ISSUE TITLE
Librispeech high WER despite long training

### ISSUE THREAD

**Issue by ErnstTmp** [tensorflow/tensor2tensor#1245]
### Description

Thanks for creating T2T!!!
I am running the lirbri speech STT for 7 days on 2 GPU (Titan X), I get a loss of approx. 0.2, a val loss of 0.27 and a target accuracy of 0.93. However, testing results look bad, and when I calculate the word error rate (WER), I get an error of 50 % (with Deep Speech 2 from Baidu, I get about 10 %), although the val loss looks good.
  
Did I do anything wrong? 
Thanks,
Ernst
### Environment information
OS: ubuntu 16.04 LTS

$ pip freeze | grep tensor

tensor2tensor==1.11.0
tensorboard==1.12.0
tensorflow-gpu==1.12.0
tensorflow-probability==0.5.0
### For bugs: reproduction and error logs
$ python -V
Python 3.5.6 :: Anaconda, Inc.
# Steps to reproduce

run thelibri speech problem with T2T
t2t-trainer \
  --model=transformer \
  --data_dir=./t2t_data \
  --output_dir=./librispeech \
  --problem=librispeech \
  --train_steps=500000 \
  --worker_gpu=2 \
  --eval_steps=3

# Error logs:
examples from testing
OUTPUT marks the predictions from the net, TARGET the expected results...

 OUTPUT: IT WAS STRENGTHENED AS OF
TARGET: ILLUSTRATION BASIL
OUTPUT: MISTER PRINCE SHOOK HIS SISTERS AND LEANED SILENT ON THE COUNTENANCE FACE AND READ THEIR DESCRIPTION RELATION CAPTAIN PETER
TARGET: MISTER PRINCE FIXED HIS AMAZED EYES ON THE CAPTAIN'S FACE AND READ THERE A DESPERATE DETERMINATION CAPTAIN HE BEGAN
OUTPUT: WE FIND WE GOT HIM TO THE LIKE THE ONE PRIZE TO BOTH ALL OTHERS AND THIS IS WHAT SHALL SHINE OR DAVIS
 TARGET: WE FINALLY GOT HIM TO SELECT THE ONE PRIZED ABOVE ALL OTHERS AND THIS IS WHAT CHEF SCHEILER GAVE US

**Comment by ErnstTmp** [comment]
Hi Vergilus, thanks, I ll try that. 
Finally, after 1.7 Mio steps, my old run seems to get somewhere, accuracy approaches 95 %, sequence accuracy is going up (from 0 to 0.16) and sentences start to look reasonable.

**Comment by vergilus** [comment]
@522730312 

```
     --problem=librispeech_clean \
     --model=transformer \
     --train_steps=500000 \
     --hparams_set=transformer_librispeech \
     --output_dir=$ASR_OUT_DIR \
     --worker_gpu=3 \
     --hparams="batch_size=2100000, \
                    optimizer="Adafactor", \
                    learning_rate_schedule="rsqrt_decay" " \
```

**Comment by veqtor** [comment]
@vergilus Is that using some sort of gradient accumulation, or are you able to fit 2.1m tokens per batch on those 3 GPU's?
How long does a step take?
